cmds/exp/nvme_unlock/nvme_unlock.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// The nvme_unlock command is used to unlock a NVMe drive with a
+// HSS-derived password, and rescan the drive to enumerate the
+// unlocked partitions.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"regexp"
+	"syscall"
+	"unsafe"
+
+	"github.com/u-root/u-root/pkg/hsskey"
+	"github.com/u-root/u-root/pkg/mount/block"
+)
+
+const (
+	opalLockUnlockIoctl = 1092120797
+)
+
+type opalKey struct {
+	lr     byte
+	keyLen byte
+	align  [6]byte
+	key    [256]byte
+}
+
+type opalSessionInfo struct {
+	sum     uint32
+	who     uint32
+	opalKey opalKey
+}
+
+type opalLockUnlock struct {
+	session opalSessionInfo
+	lState  uint32
+	align   [4]byte
+}
+
+var (
+	disk               = flag.String("disk", "/dev/nvme0n1", "The disk to be unlocked")
+	verbose            = flag.Bool("d", false, "print debug output")
+	verboseNoSanitize  = flag.Bool("dangerously-disable-sanitize", false, "Print sensitive information - this should only be used for testing!")
+	noRereadPartitions = flag.Bool("no-reread-partitions", false, "Only attempt to unlock the disk, don't re-read the partition table.")
+	lock               = flag.Bool("lock", false, "Lock instead of unlocking")
+)
+
+func verboseLog(msg string) {
+	if *verbose {
+		log.Print(msg)
+	}
+}
+
+func getSysfsInfo(index string, field string) ([]byte, error) {
+	data, err := os.ReadFile(fmt.Sprintf("/sys/class/nvme/nvme%s/%s", index, field))
+	if err != nil {
+		return nil, err
+	}
+	for i, b := range data {
+		if b == ' ' || b == '\n' {
+			return data[0:i], nil
+		}
+	}
+	return data, nil
+}
+
+func main() {
+	flag.Parse()
+
+	commandName := "unlock"
+	if *lock {
+		commandName = "lock"
+	}
+
+	diskIDRegexp := regexp.MustCompile(`/dev/nvme(\d+)n.*`)
+	diskIDMatches := diskIDRegexp.FindStringSubmatch(*disk)
+	diskID := diskIDMatches[1]
```

Comment:
`FindStringSubmatch` returns `nil` if there are no matches. Do you need to check for this value? You will encounter an error if you try to access `diskIDMatches[1]` while the slice is nil